### PR TITLE
Baseline confirmation run (new code with ood_re fix + no WD)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
We need a clean baseline run on the updated noam branch (after merging #818 ood_re fix and #849 no-weight-decay). This establishes the true noise floor for the new configuration and confirms that the improvements are reproducible.

## Instructions
Run the unmodified train.py from the current noam branch — NO changes:
```bash
python train.py --agent frieren --wandb_name "frieren/nowd-baseline" --wandb_group nowd-baseline-confirm
```

## New Baseline (run rc9qwjjv, old code)
- val/loss: 2.2396, surf_p: in_dist=20.91, ood_cond=19.71, ood_re=30.90, tandem=42.78
---
## Results

**W&B run:** `gl3gofxt` (frieren/nowd-baseline)
**Epochs:** 67 (30-min wall-clock limit)

| Split | val/loss | mae_surf_Ux | mae_surf_Uy | mae_surf_p | mae_vol_Ux | mae_vol_Uy | mae_vol_p |
|---|---|---|---|---|---|---|---|
| val_in_dist | 1.612 | 0.296 | 0.181 | **21.21** | 1.300 | 0.466 | 26.09 |
| val_ood_cond | 2.052 | 0.299 | 0.204 | **24.82** | 1.137 | 0.437 | 22.15 |
| val_ood_re | overflow | 0.279 | 0.203 | **31.62** | 1.050 | 0.443 | 51.20 |
| val_tandem_transfer | 3.267 | 0.630 | 0.342 | **42.33** | 2.114 | 0.995 | 44.26 |
| **combined val/loss** | **2.3105** | | | | | | |

**vs old-code baseline (rc9qwjjv, val/loss=2.2396):**
| Metric | Old baseline | This run | Delta |
|---|---|---|---|
| val/loss | 2.2396 | 2.3105 | **+0.071 (+3.2%)** |
| surf_p in_dist | 20.91 | 21.21 | **+0.30 (+1.4%)** |
| surf_p ood_cond | 19.71 | 24.82 | **+5.11 (+25.9%)** |
| surf_p ood_re | 30.90 | 31.62 | **+0.72 (+2.3%)** |
| surf_p tandem | 42.78 | 42.33 | **-0.45 (-1.1%)** |

**Peak GPU memory:** ~10.6 GB (estimated; system metrics not recorded)

### What happened

The new-code baseline (no WD + ood_re fix) comes in at val/loss=2.3105, which is slightly worse than the old-code baseline (2.2396). The most notable difference is ood_cond surf_p: 24.82 vs 19.71 — a +25.9% degradation. In/dist and ood_re are within ~1-2% of old baseline; tandem is slightly better.

The ood_cond surf_p spike is surprising for a run with no code changes relative to the intended baseline. A few possible explanations:

1. **Run-to-run variance**: With only 1,322 training samples, individual training runs can vary by several % in val metrics. The ood_cond split may be a stochastic outlier.
2. **No-weight-decay effect on generalization**: Removing weight decay may hurt ood_cond generalization specifically — the regularization from WD helped the model generalize to different conditions. In/dist and tandem are relatively unaffected, but ood_cond (different chord/camber ratios) may be more sensitive.
3. **Combined baseline shift**: The no-WD change legitimately shifts the noise floor. The true new baseline may be closer to 2.31 than 2.24.

The val_ood_re/loss is still showing overflow (~18869) even on this branch, which means the ood_re fix in #818 addresses filtering this out of the combined loss (correctly excluded here), not the underlying vol_p overflow at extreme Re.

### Suggested follow-ups

- **Re-run to check ood_cond variance**: A second baseline run would clarify whether the ood_cond spike is stochastic or systematic.
- **Investigate ood_re overflow**: The vol_p overflow at extreme Re is still present; the #818 fix only masks it in the combined loss. A proper fix (e.g. loss clipping or relative normalization) might unlock genuine ood_re improvements.
- **Consider WD for ood generalization**: If the no-WD change is causing ood_cond degradation, a small WD (1e-5) specifically for MLP parameters might recover generalization without hurting in-dist performance.

### CODE REVIEW

**Comment by tcapelle** [comment]
## Review — Closing
Baseline confirmation run: val_loss=2.3105 (+4.7% vs #858 baseline). The ood_cond=24.82 is an outlier (typical noise is ±5%). This confirms run-to-run variance without code changes. Closing.
